test/libraries/SafeMigration.spec.ts

```diff
@@ -0,0 +1,365 @@
+import { expect } from "chai";
+import hre, { ethers, deployments } from "hardhat";
+import {
+    getSafeWithSingleton,
+    getSafeSingletonAt,
+    safeMigrationContract,
+    getSafeSingletonContract,
+    getSafeL2SingletonContract,
+    getCompatFallbackHandler,
+} from "../utils/setup";
+import deploymentData from "../json/safeDeployment.json";
+import fallbackHandlerDeploymentData from "../json/fallbackHandlerDeployment.json";
+
+import { executeContractCallWithSigners } from "../../src/utils/execution";
+import { SafeMigration } from "../../typechain-types";
+
+const FALLBACK_HANDLER_STORAGE_SLOT = "0x6c9a6c4a39284e37ed1cf53d337577d14212a4870fb976a4366c693b939918d5";
+
+const GUARD_STORAGE_SLOT = "0x4a204f620c8c5ccdca3fd54d003badd85ba500436a431f0cbda4f558c93c34c8";
+
+const migrationPaths = [
+    {
+        testSuiteName: "1.3.0 to latest (1.5.0)",
+        from: { safeDeploymentData: deploymentData.safe130, safeL2DeploymentData: deploymentData.safe130l2 },
+        latest: true,
+    },
+    {
+        testSuiteName: "1.3.0 to 1.4.1",
+        from: { safeDeploymentData: deploymentData.safe130, safeL2DeploymentData: deploymentData.safe130l2 },
+        // `to` is used when the target contracts are not latest. So, we need to provide the bytecode
+        // of the old version of the contracts
+        to: {
+            safeDeploymentData: deploymentData.safe141,
+            safeL2DeploymentData: deploymentData.safe141l2,
+            safeCompatFallbackHandler: fallbackHandlerDeploymentData.compatibilityFallbackHandler141,
+        },
+    },
+    {
+        testSuiteName: "1.4.1 to latest (1.5.0)",
+        from: {
+            safeDeploymentData: deploymentData.safe141,
+            safeL2DeploymentData: deploymentData.safe141l2,
+        },
+        latest: true,
+    },
+];
+
+describe("SafeMigration Library", () => {
+    const migratedInterface = new ethers.Interface(["function masterCopy() view returns(address)"]);
+
+    let SAFE_SINGLETON_ADDRESS: string | null | undefined;
+    let SAFE_SINGLETON_L2_ADDRESS: string | null | undefined;
+    let COMPATIBILITY_FALLBACK_HANDLER: string | null | undefined;
+    let migration: SafeMigration;
```

Comment:
This isn't needed at this scope, it is returned by the `setupTests`.

Reply:
https://github.com/safe-global/safe-smart-account/pull/793/commits/15bdca03f4590d44c926d9ffd779adea6eac5f27